### DISCUSSION TITLE
disable Login from a new location emails

### DISCUSSION THREAD

**Discussion by orenaksakal** [pocketbase/pocketbase#5942] [Q&A] [answered]
I've upgraded to 0.23.0 successfully but currently being bombarded by automated `Login from a new location` emails 

<img width="1129" alt="Screenshot 2024-11-24 at 3 17 21 PM" src="https://github.com/user-attachments/assets/32851121-b58a-4e8e-b941-e33132178a4d">

I've unchecked this in admin settings but seems like its not effective

<img width="694" alt="Screenshot 2024-11-24 at 3 17 44 PM" src="https://github.com/user-attachments/assets/196d8d83-228c-4bf2-b39b-2c2f83c3fa3d">

Is there a way to disable this email?

**Comment by ganigeorgiev** [comment, marked as answer]
I'm not able to reproduce it.

There are also automated tests for both the internal helper method and the auth-with-password endpoint that looks correct to me:
- https://github.com/pocketbase/pocketbase/blob/master/apis/record_helpers_test.go#L340
- https://github.com/pocketbase/pocketbase/blob/master/apis/record_auth_with_password_test.go#L368

Are you sure that the collection option change was actually persisted and it was done for the correct auth collection?

Are you sure that the received emails were not just delayed from before changing the option (_sometimes mail providers may keep the emails in a queue before processing them_)?

Technically you can attach to the [`OnMailerRecordAuthAlertSend`](https://pocketbase.io/docs/go-event-hooks/#onmailerrecordauthalertsend) hook and return `nil` instead of calling `e.Next()` but this shouldn't be needed in your case and I'd advice finding out why the toggle doesn't work.

Either you are not testing properly (e.g. changing the settings of a different PocketBase instance or collection) or there is something else strange going on with your environment (_I remember you from https://github.com/pocketbase/pocketbase/discussions/5611 so if you are using a custom SQLite driver make sure that it is not caused by it_).

If you are still not able to identify the culprit, please export your collections configuration as json (Dashboard > Settings > Export collections) and elaborate a little more on your setup (do you use the prebuilt executable, custom driver, with what collection you are authenticating, etc.).

**Reply by orenaksakal** [reply]
I think I figured what triggers these and its crazy; It happens when I deploy a new version of my sveltekit app. 
I use Cloudflare workers and there is a `admin.ts` file which I use to initialize admin client like

```javascript
import { SECRET_POCKETBASE_EMAIL, SECRET_POCKETBASE_PW } from '$env/static/private';
import { PUBLIC_POCKETBASE_URL } from '$env/static/public';
import type { TypedPocketBase } from '$lib/pocketbase/types';
import PocketBase from 'pocketbase';

export const pocketbaseAdminClient = new PocketBase(PUBLIC_POCKETBASE_URL) as TypedPocketBase;

pocketbaseAdminClient.autoCancellation(false);

await pocketbaseAdminClient
	.collection('_superusers')
	.authWithPassword(SECRET_POCKETBASE_EMAIL, SECRET_POCKETBASE_PW, {
		// This will trigger auto refresh or auto reauthentication in case
		// the token has expired or is going to expire in the next 30 minutes.
		autoRefreshThreshold: 30 * 60
	});

```

this is executed on each deploy counting the login from cloudflare builder as a new login, its also visible in the logs

<img width="1645" alt="Screenshot 2024-11-24 at 7 23 37 PM" src="https://github.com/user-attachments/assets/b3396f9f-b71c-4bec-ae0b-a880ed3dd806">

**Reply by orenaksakal** [reply]
also reproducable with my webhook where I use the same `pocketbaseAdminClient` from above

**Reply by orenaksakal** [reply]
I had to disable the toggle for `_superusers` as well. It's all good now TY 👍
